src/storage.rs

```diff
@@ -1,12 +1,38 @@
-use crate::prelude::{Address, H256};
+use crate::prelude::{Address, Vec, H256};
+use borsh::{BorshDeserialize, BorshSerialize};
 
 #[allow(dead_code)]
+#[derive(Clone, Copy, BorshSerialize, BorshDeserialize)]
 pub enum KeyPrefix {
     Config = 0x0,
     Nonce = 0x1,
     Balance = 0x2,
     Code = 0x3,
     Storage = 0x4,
+    RelayerEvmAddressMap = 0x5,
+}
+
+/// We can't use const generic over Enum, but we can do it over integral type
+pub type KeyPrefixU8 = u8;
+
+// TODO: Derive From<u8> using macro to avoid missing new arguments in the future
```

Comment:
If this doesn't get resolved in the larger connector effort then we should file an issue for it. I'm not a big fan of this brittle-looking match. Maybe we should have a unit test that round-trips all `KeyPrefix` variants through `u8` and back to make sure we don't hit the `unreachable`.

Would it be too much overhead to use borsh here instead of writing the match explicitly?

Reply:
> Would it be too much overhead to use borsh here instead of writing the match explicitly?

Using borsh might not be a good idea, since I believe borsh will see those fields as `isize` so the encoding will have 5 bytes, and will incur in some (not big but) unnecessary overhead.

I'll add the macro to derive this in separate PR to merge this now.